### PR TITLE
chore: capture performance of parsing api response

### PR DESCRIPTION
**Pull request by vsvamsi1**
## Description
Capture performance of parsing api responses.

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11512264036>
> Commit: 92f9d882b4f0dd40b1b3e38cb817b371b8288a68
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11512264036&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Fri, 25 Oct 2024 06:47:52 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced a telemetry feature for enhanced monitoring of API requests and responses.

- **Bug Fixes**
	- Improved error handling by logging issues when default transformation functions are not found.

### CODE REVIEW

File: app/client/src/api/Api.ts [hunk 1]
```diff
@@ -19,6 +20,30 @@ export const apiRequestConfig = {
...
+    const transformResponseAr = axios.defaults.transformResponse;
+
+    // Pick up the transformFn from the defaults and wrap it in with telemetry code
+    if (Array.isArray(transformResponseAr) && transformResponseAr?.[0]) {
+      const transfromFn = transformResponseAr?.[0];
+      const resp = startAndEndSpanForFn(
+        "transformApiResponse",
+        { url: this.url },
+        transfromFn.call(this, ...args),
+      );
```
[the remark this answers is not in the document]
Comment (dvj1988) [reply app/client/src/api/Api.ts:34]: @vsvamsi1 can you check this comment?

Call will execute the function in the same statement. Did you mean to use bind here?

File: app/client/src/api/Api.ts [hunk 2]
```diff
@@ -19,6 +20,30 @@ export const apiRequestConfig = {
...
 
+axiosInstance.defaults.transformResponse = [
+  function (...args) {
+    const transformResponseAr = axios.defaults.transformResponse;
+
+    // Pick up the transformFn from the defaults and wrap it in with telemetry code
+    if (Array.isArray(transformResponseAr) && transformResponseAr?.[0]) {
+      const transfromFn = transformResponseAr?.[0];
+      const resp = startAndEndSpanForFn(
+        "transformApiResponse",
```
Comment (dvj1988) [inline app/client/src/api/Api.ts:31]: I think we should start having a naming convention for all new spans. 
Can we name it something like `axios. transformApiResponse`?

Comment (dvj1988) [reply app/client/src/api/Api.ts:31]: cc: @vsvamsi1

File: app/client/src/api/Api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vsvamsi1) [reply app/client/src/api/Api.ts]: @dvj1988 My bad that was a miss, yes I wanted to execute just not bind.

**Review by dvj1988 — APPROVED** [review]

**Pull request merged**